### ISSUE TITLE
List info_keys (for dataset's alias) is too short for model development use case 

### ISSUE THREAD

**Issue by senesis** [ESMValGroup/ESMValCore#1144]
When using ESMValTool for model development workflow, the 'alias' for a dataset (which is used for a label in graphics, and is computed in class Recipe) is not informative enough, because it is based on a fixed list of attributes , `info_keys`, which value is : `'project', 'activity', 'dataset', 'exp', 'ensemble', 'version'` 

These keys are not enough if the project which describes that model's data chooses to use keywords that are meaningful to the community of developers for that model, rather than disguise it using 'ensemble' or 'version' keywords

A use case is the set of IPSLCM standard labels used in it DRS :
```
  input_dir:
    default : '{account}/{model}/{status}/{exp}/{simulation}/{igcm_dir}/Analyse/{freq}' 

```
which should presently be turned, e.g. to :
```
  input_dir:
    default : '{activity}/{model}/{version}/{exp}/{ensemble}/{igcm_dir}/Analyse/{freq}' 

```
for matching the `info_keys` list. This would be uneasy to understand and to use in recipe datasets section, and still be not enough for translating keyword `model` (because remaining keys in `info_keys`, i.e. : `project` and `dataset` are fixed for this data source). This applies for cases where a recipe is used for comparing multiple datasets from the model under development.

A solution could be that the project or dataset definition includes a section indicating which keys should be added to the default `info_keys` list. This would have to be applicable whatever the choices made in #494  

[ Pinging individuals involved in model development use case : @zklaus @bouweandela @alistairsellar  ]

By the way : I searched the doc for `alias` and didn't find any mention for the possibility to customize a dataset's alias using keyword `alias`

**Comment by bouweandela** [comment]
Would it also work to just use whatever keys are available for a particular dataset in alphabetical order if the default keys do not offer sufficient differentiation? I would prefer to keep it simple and not add too much custom code for non-CMIP data. The alias function was implemented by @jvegasbsc, maybe he has an opinion?

**Comment by senesis** [comment]
> Would it also work to just use whatever keys are available for a particular dataset in alphabetical order if the default keys do not offer sufficient differentiation?

Yes, that may work, assuming that 'whatever keys are available' is understood as 'whatever key are used in input_dir and input_file patterns'
